geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearWithRebalanceDUnitTest.java

```diff
@@ -0,0 +1,376 @@
+      Assertions.assertThat(partitionedRegionStats.getVolunteeringInProgress()).isEqualTo(0);
+      Assertions.assertThat(partitionedRegionStats.getBucketCreatesInProgress()).isEqualTo(0);
+      Assertions.assertThat(partitionedRegionStats.getPrimaryTransfersInProgress()).isEqualTo(0);
+      Assertions.assertThat(partitionedRegionStats.getRebalanceBucketCreatesInProgress())
+          .isEqualTo(0);
+      Assertions.assertThat(partitionedRegionStats.getRebalancePrimaryTransfersInProgress())
+          .isEqualTo(0);
+    });
+  }
+
+  private void assertRegionIsEmpty(List<VM> vms) {
+    vms.forEach(vm -> vm.invoke(() -> {
+      waitForSilence();
+      PartitionedRegion region = (PartitionedRegion) cacheRule.getCache().getRegion(REGION_NAME);
+
+      Assertions.assertThat(region.getLocalSize()).isEqualTo(0);
+    }));
+  }
+
+  private void registerVMKillerAsCacheWriter(List<VM> vmsToBounce) {
+    vmsToBounce.forEach(vm -> vm.invoke(() -> {
+      Region<String, String> region = cacheRule.getCache().getRegion(REGION_NAME);
+      region.getAttributesMutator().setCacheWriter(new MemberKiller());
+    }));
+  }
+
+  @Test
+  @Parameters(method = "vmsAndRegionTypes")
+  @TestCaseName("[{index}] {method}(ClearCoordinator:{0}, RebalanceCoordinator:{1}, RegionType:{2})")
+  public void clearRegionDuringRebalanceClearsRegion(TestVM clearCoordinatorVM,
+      TestVM rebalanceVM, RegionShortcut regionType) throws InterruptedException {
+    AsyncInvocation<?> clearInvocation = setupAndPrepareClear(clearCoordinatorVM, regionType);
+
+    getVM(rebalanceVM.vmNumber).invoke(() -> {
+      RebalanceResults results = startRebalanceAndGetResults();
+
+      // Verify that rebalance did some work
+      int combinedResults = results.getTotalBucketTransfersCompleted()
+          + results.getTotalBucketCreatesCompleted() + results.getTotalPrimaryTransfersCompleted();
+      assertThat(combinedResults, greaterThan(0));
+
+      // Verify that no bucket creates failed during the rebalance
+      assertThat(cacheRule.getCache().getInternalResourceManager().getStats()
+          .getRebalanceBucketCreatesFailed(), is(0));
+    });
+
+    clearInvocation.await();
+
+    // Assert that the region is empty
+    assertRegionIsEmpty(asList(accessor, server1, server2));
+  }
+
+  @Test
+  @Parameters(method = "vmsAndRegionTypes")
+  @TestCaseName("[{index}] {method}(ClearCoordinator:{0}, RebalanceCoordinator:{1}, RegionType:{2})")
+  public void clearRegionDuringRebalancePrimaryReassignmentClearsRegion(TestVM clearCoordinatorVM,
+      TestVM rebalanceVM, RegionShortcut regionType) throws InterruptedException {
+    AsyncInvocation<?> clearInvocation = setupAndPrepareClear(clearCoordinatorVM, regionType);
+
+    getVM(rebalanceVM.vmNumber).invoke(() -> {
+      // Start a rebalance and wait until primary reassignment has started before signalling the
+      // blackboard
+      RebalanceOperation rebalanceOp =
```

Comment:
Its same as "startRebalanceAndGetResults()"; can it be called here as done in previous test.

Reply:
This case is actually slightly different; it waits for `getRebalance**PrimaryTransfers**Completed()` to be greater than zero instead of `getRebalance**BucketCreates**Completed()`. This case is intended to test the behaviour of clear when rebalance is doing the reassigning primaries step specifically.